ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py

```diff
@@ -206,7 +206,19 @@ def get_column_tag_labels(
         return None
 
     def set_inspector(self, database_name: str):
-        self.client = Client(project=database_name)
+        # TODO support location property in JSON Schema
+        # TODO support OAuth 2.0 scopes
+        impersonate_service_account = (
+            self.service_connection.credentials.gcpImpersonateServiceAccount.impersonateServiceAccount
+        )
+        lifetime = (
+            self.service_connection.credentials.gcpImpersonateServiceAccount.lifetime
+        )
+        self.client = get_bigquery_client(
+            project_id=database_name,
+            impersonate_service_account=impersonate_service_account,
+            lifetime=lifetime,
+        )
```

Comment:
I have improved this by getting properties through the object. 

I plan to support other properties in another pull request. One is `location` to specify a location of a GCP client. The other is to specify OAuth 2.0 scopes. For instance, if we have BigQuery external tables whose sources are located in Google Drive, we have to pass `https://www.googleapis.com/auth/drive` in addition to `https://www.googleapis.com/auth/cloud-platform`. It would be good to pass such optional information to no only the GCP credentials but also the target impersonate service account.